A73-transport-selection.md

```diff
@@ -0,0 +1,119 @@
+Transport Selection
+----
+* Author(s): dfawley
+* Approver: roth, ejona86
+* Status: Draft
+* Implemented in: none
+* Last updated: 2023-11-07
+* Discussion at: TODO: TBD
+
+## Abstract
+
+A standardized method to dynamically choose between multiple transports.
+
+## Background
+
+gRPC predominately uses HTTP/2 over TCP to communicate between client and
+server. In some implementations, there are existing methods for circumenting
+this to instead use an in-process transport. There are also use cases that would
+like to take advantage of alternate transports, e.g.
+[QUIC+HTTP/3](https://github.com/grpc/proposal/blob/master/G2-http3-protocol.md)
+and a way of communicating over shared memory (design TBD).
+
+### Related Proposals: 
+
+* [L37: Go Custom Transports](https://github.com/grpc/proposal/pull/103)
+  (pending)
+* [G2: gRPC over
+  HTTP/3](https://github.com/grpc/proposal/blob/master/G2-http3-protocol.md)
+* [A6: Client
+  Retries](https://github.com/grpc/proposal/blob/master/A6-client-retries.md)
+
+## Proposal
+
+This proposal is for a standard way of selecting which transport to use in gRPC
+clients and servers. The exact details will be left up to gRPC implementations
+to decide; only the high-level method is included in this design.
+
+Some additional design goals of this proposal:
```

Comment:
This is awkward. It is in the proposal section, and defines a mix of goals and design choices. I think "goals" isn't the right word here, as these are part of the proposal itself. Goals shouldn't be in "Proposal" anyway.

Reply:
Moved into a sub-section about transports.